src/dotnet/APIView/APIView/AssemblyAPIV.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace APIView
+{
+    /// <summary>
+    /// Class representing a C# assembly. Each assembly has a name and global namespace, 
+    /// which may or may not contain further types.
+    /// 
+    /// AssemblyAPIV is an immutable, thread-safe type.
+    /// </summary>
+    public class AssemblyAPIV
+    {
+        public string Name { get; }
+        public NamespaceAPIV GlobalNamespace { get; }
+
+        /// <summary>
+        /// Construct a new AssemblyAPIV instance, represented by the provided symbol.
+        /// </summary>
+        /// <param name="symbol">The symbol representing the assembly.</param>
+        public AssemblyAPIV(IAssemblySymbol symbol)
+        {
+            this.Name = symbol.Name;
+            this.GlobalNamespace = new NamespaceAPIV(symbol.GlobalNamespace);
+        }
+
+        public static List<AssemblyAPIV> AssembliesFromFile(string dllPath)
+        {
+            var compilation = GetCompilation(dllPath);
+            List<AssemblyAPIV> assemblies = new List<AssemblyAPIV>();
+
+            foreach (var assemblySymbol in compilation.SourceModule.ReferencedAssemblySymbols)
+            {
+                assemblies.Add(new AssemblyAPIV(assemblySymbol));
```

Comment:
Would this add all trusted assemblies to the `assemblies` not just the `dllPath`?

Reply:
It does - currently for testing I return all assemblies and select the one with the desired name, but I should be able to update this to only return the specified assembly.